docs/recipes/LocalFirstDataWithPowerSync.md

```diff
@@ -0,0 +1,713 @@
+            new Column({name: 'title', type: ColumnType.TEXT}),
+            new Column({name: 'completed', type: ColumnType.BOOLEAN}),
+            // Add other columns as needed
+        ],
+    }),
+    // Define other tables as needed
+]);
+
+// It's a good idea to export typescript types for your schema as well, and keep them near the schema types they describe.
+export interface Todo {
+    id: string;
+    title: string;
+    completed: boolean;
+}
+```
+
+:::tip Automated Schema Generation
+
+PowerSync can generate a Javascript version of your schema for you. To do this, run the following command in your
+project directory.
+
+1. Right click on your instance in the PowerSync dashboard
+2. Select "Generate Client-Side Schema".
+
+This will generate a `schema.js` file which can still save you a lot of time as you get started.
+:::
+
+## Setting Up the Database Connection and Context
+
+Next we need to integrate PowerSync and Supabase by setting up the database connection, and then providing that connection to the app through a React Context.. 
+
+### Connecting to Your Database
+
+To tell PowerSync how to connect to the database we'll create a `PowerSyncBackendConnector` that
+has methods for fetching credentials and uploading data.
+
+#### PowerSyncBackendConnector Interface
+
+The Supabase Connector needs to implement the `PowerSyncBackendConnector` interface, ensuring it can seamlessly
+communicate with PowerSync for data synchronization. 
+
+The interface requires two methods:
+
+```ts
+// fetches credentials for connecting to the backend
+fetchCredentials: () => Promise<PowerSyncCredentials | null>;
+
+// a method powersync will call to upload data to the backend
+uploadData: (database: AbstractPowerSyncDatabase) => Promise<void>;
+```
+
+The Powersync docs provide [detailed instructions](https://docs.powersync.com/client-sdk-references/react-native-and-expo#id-2.-create-a-backend-connector) on how to implement the `PowerSyncBackendConnector` interface.
+
+#### Implementing a Database Connector for Supabase
+
+We Initialize the Supabase client with session persistence and custom storage for persisting keys. 
+
+Persisting the Supabase session is essential for maintaining user sessions across app restarts. Unlike web environments
+where `localStorage` is available, React Native requires a different approach for secure storage.
+
+The`react-native-supabase-todolist` demo from PowerSync has a [simple implementation of a kv-store](https://github.com/powersync-ja/powersync-js/blob/main/demos/react-native-supabase-todolist/library/storage/SupabaseStorageAdapter.ts)
+that works well for this. (imported below as `KVStorage`)
+
+```ts
+// app/services/database/supabase.ts
+import {
+    AbstractPowerSyncDatabase,
+    PowerSyncBackendConnector,
+    UpdateType
+} from "@journeyapps/powersync-sdk-react-native";
+import {createClient} from "@supabase/supabase-js";
+import Config from "app/config";
+import {KVStorage} from "./kv-storage";
```

Comment:
~Am I supposed to grab this from somewhere?~

The link above was wrong. Now what's interesting is that I also need to install `expo-secure-storage` to make this work.